src/components/listing-components/listing-screen-post.tsx

```diff
@@ -0,0 +1,155 @@
+import jwt_decode from "jwt-decode";
+import strings from "localization/strings";
+
+/**
+ * Interface for POST item
+ */
+interface Item {
+  title: string;
+  expired: boolean;
+  category: string;
+  onlyForCompanies: boolean,
+  metadata: { locationInfo: { phone: string; address: string; email: string; } };
+  images?: string[];
+  properties?: { key: string; value: string }[];
+  userId: string,
+  price: number;
+  priceUnit: string;
+  paymentMethod: string,
+  delivery: boolean,
+  itemType: "SELL";
+}
+
+/**
+ * Interface representing a decoded access token
+ */
+interface DecodedAccessToken {
+  sub?: string;
+  given_name?: string;
+  family_name?: string;
+  created?: Date;
+  userId?: string;
+  access_token?: string;
+  refresh_token?: string;
+  refresh_expires_in?: number;
+}
+
+/**
+ * interface for params itemdata
+ */
+interface ItemData {
+  title: string;
+  category: string;
+  address: string;
+  email: string;
+  phone: string;
+  description: string;
+  unit: "KG" | "TN" | "M2" | "M3" | "PCS";
+  amount: number;
+  price: number;
+}
+
+/**
+ * Builds access token object from login data
+ *
+ * @param tokenData token data
+ * @returns access token
+ */
+const buildToken = (tokenData: any) => {
+  const decodedToken: DecodedAccessToken = jwt_decode(tokenData.access_token);
+  const created = new Date();
+
+  return {
+    created: created,
+    access_token: tokenData.access_token,
+    expires_in: tokenData.expires_in,
+    refresh_token: tokenData.refresh_token,
+    refresh_expires_in: tokenData.refresh_expires_in,
+    userId: decodedToken.sub
+  };
+};
+
+/**
+ * Handle material unit type and update itemData properties accordingly
+ * Change material unit type to match Kiertoon.fi specification
+ * 
+ * @param data 
+ * @param itemData 
+ */
+const handleMaterialUnit = (data: any, itemData: Item) => {
+  if (data.unit === "KG") {
+    itemData.properties?.push({ key: "Paino", value: `${data.amount} ${data.unit}` });
+  } else if (data.unit === "TN") {
+    itemData.properties?.push({ key: "Paino", value: `${data.amount} ${data.unit}` });
+  } else if (data.unit === "M2") {
+    itemData.properties?.push({ key: "Tilavuus", value: `${data.amount} ${data.unit}` });
+  } else if (data.unit === "M3") {
+    itemData.properties?.push({ key: "Tilavuus", value: `${data.amount} ${data.unit}` });
+  } else if (data.unit === "PCS") {
+    itemData.properties?.push({ key: "Määrä", value: `${data.amount} KPL` });
+  }
+};
```

Comment:
```suggestion
/**
 * Get item properties
 * 
 * @param data item data
 * @returns properties 
 */
const getItemProperties = (data: ItemData) => {
  const { unit, amount, description } = data;
  const properties: { key: string; value: string }[] = [];
  const unitAmountString = `${amount} ${unit}`;

  switch (unit) {
    case "KG":
      properties.push({ key: "Paino", value: unitAmountString });
      break;
    case "TN":
      properties.push({ key: "Paino", value: unitAmountString });
      break;
    case "M2":
      properties.push({ key: "Tilavuus", value: unitAmountString });
      break;
    case "M3":
      properties.push({ key: "Tilavuus", value: unitAmountString });
      break;
    case "PCS":
      properties.push({ key: "Määrä", value: `${amount} KPL` });
      break;
    default:
      break;
  }

  properties.push({ key: "Lisätiedot", value: description || "" });

  return properties;
};
```

Reply:
Refactored this a little bit so we can use one function to return all the needed properties. Please test the functionality after commit.

Reply:
And the reason for this change is that the previous way was mutating the parameter without returning it. That is not advicable even though javascript supports such behaviour.